spec/timezone.html

```diff
@@ -191,6 +191,7 @@ <h1>get Temporal.TimeZone.prototype.id</h1>
       <emu-alg>
         1. Let _timeZone_ be the *this* value.
         1. Perform ? RequireInternalSlot(_timeZone_, [[InitializedTemporalTimeZone]]).
+        1. If _timeZone_.[[OffsetNanoseconds]] is not ~empty~, return FormatTimeZoneOffsetString(_timeZone_.[[OffsetNanoseconds]]).
```

Comment:
this seems normative - will this return something equivalent to the ID?

Reply:
Unless I'm missing something, I believe this is an editorial change.

In current spec text, offset time zone IDs are canonicalized in [CreateTemporalTimeZone](https://tc39.es/proposal-temporal/#sec-temporal-createtemporaltimezone) which is called by the TimeZone constructor.

So in current spec text, it should not be possible to end up with an offset time zone `_timeZone_` whose `[[Identifier]]` slot is different from the result of `FormatTimeZoneOffsetString(_timeZone_.[[OffsetNanoseconds]])`. 

What's the normative change here?

Reply:
ah ok thanks, that wasn’t clear to me. If the information is already cached/precomputed, why include another AO call here?

Reply:
This PR tries to clarify that this string doesn't need to be precomputed. The current spec text implies that a TimeZone's storage requirements are: a string which can be as long as 34 characters for an IANA zone or 20 characters for an offset time zone AND a 47-bit integer representing the ±8.64e13 offset. If allocating on the stack, that's 40 bytes!

The actual TimeZone storage requirement is 6 bytes: the union of a 10-bit index into the static list of ~600 IANA identifiers OR a 47-bit integer representing the ±8.64e13 offset.

The goal of this PR is to clarify the actual storage requirements. One of the concerns we've heard from implementers,  most recently at the last TG2 meeting, is that Temporal objects in general (and time zones in particular) will be too large in memory.

In some cases (like the pending change to put bounds on Duration fields) there's normative changes required to reduce storage requirements. But thankfully in the TimeZone case we just need to clarify the spec. Or at least that's what I think is all we need to do! :-) 

Implementers may decide to pay the RAM tax in exchange for not having to build ID strings on the fly, but IMO it's a good idea to be more clear in the spec that precomputing is optional and they don't have to precompute and store the ID string before it's needed.

Reply:
Wouldn't an assertion clarify that without implying the need for another AO call?

Reply:
Jordan and I synced offline about this, and we agreed to add another paragraph to the note under `CreateTemporalTimeZone`:

```
<p>
  Similar flexibility exists for the storage of the [[OffsetNanoseconds]] internal slot, which can be interchangeably represented as a 6-byte integer or as a String value that may be as long as 19 characters.
  ParseTimeZoneOffsetString and FormatTimeZoneOffsetString may be used to losslessly convert one representation to the other.
  Implementations are free to store either or both representations.
</p>
```